pygments/lexers/mumps.py

```diff
@@ -0,0 +1,915 @@
+"""
+    pygments.lexers.mumps
+    ~~~~~~~~~~~~~~~~~~~~~
+
+    Lexers for MUMPS
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import re
+from pygments.lexer import ExtendedRegexLexer, words, bygroups, default, include
+from pygments.token import Whitespace, Keyword, Operator, Number, Name, Punctuation, Comment, String
+
+__all__ = ['MumpsLexer']
+
+class MumpsLexer(ExtendedRegexLexer):
+    """
+    For MUMPS source code.
+
+    Derived from `The Annotated M[UMPS] Standard <http://71.174.62.16/Demo/AnnoStd>`
+    Section numbers below refer to the sections from the 1995 (current) standard on that site
+    """
+
+    name = 'MUMPS'
+    aliases = ['mumps', 'm']
+    filenames = ['*.m', '*.mumps', '*.epc', '*.int']
+    # Filenames aren't meaningful in M, but some implementations allow file export/import of routines
+    # For example, YottaDB would have the source file for "dmex" in "dmex.m"
+    flags = re.IGNORECASE
+
+    # 5 - Metalanguage description
+    # 'L': list of one or more, separated by commas
+    def L(state):
+        return [default(('list_comma', state))]
+
+    # Definitions of groups that we implement as regular expressions
+    # 6.1 - Routine head 'routinehead'
+    name_re = '[%A-Za-z][A-Za-z0-9]*'
+    # 6.2.4 - Label separator 'ls'
+    ls_re = ' +'
+    # 7.1.4.1 - String literal 'strlit'
+    strlit_re = '"(""|[^"])*"'
+    # 7.1.4.10 - Intrinsic special variables 'svn'
+    svn_re = words(('$DEVICE', '$D', '$ECODE', '$EC', '$ESTACK', '$ES', '$ETRAP', '$ET', '$HOROLOG', '$H', '$IO', '$I', '$JOB', '$J', '$KEY', '$K', '$PRINCIPAL', '$P', '$QUIT', '$Q', '$STACK', '$ST', '$STORAGE', '$S', '$SYSTEM', '$SY', '$TEST', '$T', '$TLEVEL', '$TL', '$TRESTART', '$TR', '$X', '$Y'), suffix=r'\b(?!\()')
+    # 7.1.4.11 - Unary operator 'unaryop'
+    unaryop_re = "[-+']"
+    # 7.2.1 - binaryop
+    binaryop_re = r'\*\*|[-_+*/\\#]'
+    # 7.2.2 - truthop
+    relation_re = r'\]\]|[<>=\[\]]' # 7.2.2.1 - Relational operator 'relation'
+    logicalop_re = '[&!]' # 7.2.2.4 - Logical operator 'logicalop'
+    truthop_re = relation_re + '|' + logicalop_re
+
+    def analyse_text(text):
+        if re.search(r'^[%A-Za-z][A-Za-z0-9]* ;', text):
+            return 0.3
+        if re.search(r'^[%A-Za-z][A-Za-z0-9]*\n', text):
+            return 0.1
```

Comment:
Aren't there some more specific ways to recognize MUMPS? There patterns are very unspecific, it seems to me that they will trigger on many languages ...

Reply:
True, I will work on finding some patterns that have better matching. The joys of old terse languages that existed on dedicated machines...